front/components/assistant_builder/TeamSharingSection.tsx

```diff
@@ -9,74 +10,152 @@ import {
+
+  const [requestNewScope, setModalNewScope] = useState<Exclude<
+    AgentConfigurationScope,
+    "global"
+  > | null>(null);
   const agentUsage = useAgentUsage({
     workspaceId: owner.sId,
     agentConfigurationId,
   });
 
+  const usageText = assistantName
+    ? assistantUsageMessage({
+        assistantName,
+        usage: agentUsage.agentUsage,
+        isLoading: agentUsage.isAgentUsageLoading,
+        isError: agentUsage.isAgentUsageError,
+      })
+    : "";
+
   const scopeInfo: Record<
     Exclude<AgentConfigurationScope, "global">,
     {
       label: string;
       color: string;
       icon: typeof UserGroupIcon | typeof PlanetIcon | typeof LockIcon;
       text: string;
+      confirmationModalData: ConfirmationModalDataType;
     }
   > = {
     published: {
       label: "Shared Assistant",
       color: "pink",
       icon: UserGroupIcon,
       text: "Anyone in the workspace can view and edit.",
+      confirmationModalData: {
+        title: "Sharing an assistant",
+        text: "Once shared, the assistant will be visible and editable by members of your workspace.",
+        confirmText: "Share the assistant",
+        variant: "primary",
+      },
     },
     workspace: {
       label: "Company Assistant",
       color: "amber",
       icon: PlanetIcon,
       text: "Activated by default for all members of the workspace.",
+      confirmationModalData: {
+        title: "Moving to Company Assistants",
+        text: "Moving the assistant to Company Assistants will make the assistant editable only by Admins and Builders and add it the “My Assistants” list of all the workspace.",
+        confirmText: "Move to Company",
+        variant: "primary",
+      },
     },
     private: {
       label: "Personal Assistant",
       color: "sky",
       icon: LockIcon,
       text: "Only I can view and edit.",
+      confirmationModalData: {
+        title: "Moving to Personal Assistants",
+        text: `Moving the assistant to your Personal Assistants will make the assistant unaccessible to other members of the workspace.`,
+        confirmText: "Move to Personal",
+        variant: "primaryWarning",
+        usageText,
+      },
     },
   };
-
+  // special case if changing setting from company to shared
+  const companyToSharedModalData: ConfirmationModalDataType = {
+    title: "Moving to Shared Assistants",
+    text: `Moving ${
+      assistantName || "the assistant"
+    } to Shared Assistants will make the assistant editable by all members of the workspace and the assistant will not be activated by default anymore.`,
+    confirmText: "Move to Shared",
+    variant: "primary",
+    usageText,
+  };
   return (
     <div className="flex flex-col gap-3">
       <div className="text-lg font-bold text-element-900">Sharing</div>
       <div>
+        {requestNewScope && (
+          <ScopeChangeModal
+            show={requestNewScope !== null}
+            confirmationModalData={
+              requestNewScope
```

Comment:
Can we extract this part and avoid nested ternaries?

Reply:
we can; I agree it's not optimal;  but other options don't seem very good too (if requires a let + null check etc). Do you see a way of writing this well & concisely?